File: two-sum/python/test_two_sum.py
from two_sum import two_sum

if __name__ == "__main__":
    # Test 1: basic case
    result = two_sum([2, 7, 11, 15], 9)
    assert result == [0, 1], f"Expected [0, 1], got {result}"
    print(f"Python: two_sum([2,7,11,15], 9) = {result} — PASS")

    # Test 2: no solution
    result = two_sum([1, 2, 3], 10)
    assert result == [], f"Expected [], got {result}"
    print(f"Python: two_sum([1,2,3], 10) = {result} — PASS")

    # Test 3: duplicate values
    result = two_sum([3, 3], 6)
    assert result == [0, 1], f"Expected [0, 1], got {result}"
    print(f"Python: two_sum([3,3], 6) = {result} — PASS")

    print("All Python tests passed!")

File: two-sum/javascript/test_twoSum.js
const { twoSum } = require('./twoSum');

function assertEqual(actual, expected, label) {
    const pass = JSON.stringify(actual) === JSON.stringify(expected);
    console.log(`JavaScript: ${label} — ${pass ? 'PASS' : 'FAIL'}`);
    if (!pass) {
        console.error(`  Expected: ${JSON.stringify(expected)}, got: ${JSON.stringify(actual)}`);
        process.exit(1);
    }
}

// Test 1: basic case
assertEqual(twoSum([2, 7, 11, 15], 9), [0, 1], 'twoSum([2,7,11,15], 9)');

// Test 2: no solution
assertEqual(twoSum([1, 2, 3], 10), [], 'twoSum([1,2,3], 10)');

// Test 3: duplicate values
assertEqual(twoSum([3, 3], 6), [0, 1], 'twoSum([3,3], 6)');

console.log('All JavaScript tests passed!');
